src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -666,6 +666,11 @@ public CitizenData getCitizenData()
     @Override
     public boolean hasCapability(final Capability<?> capability, final EnumFacing facing)
     {
+        if (getCitizenData() == null)
+        {
+            return false;
```

Comment:
I suggest we do `return super.hasCapability(capability, facing);` here to not block behaviour of subclasses.

Reply:
No we should not. We do not support Item_Handler capability if we do not have the Entity. If we would use super here, we would need to use super in the getCap method too. Which could leed to undefined behaviour.

Simply said, the question here is can we access the citizens inventory yes or no. If not then return false.

Reply:
heh, i just applied the same change to `getCapability` with super, since it was still able to fail
so we can either turn it back to false, or keep the current diff?

Reply:
What i mean is that the behaviour of super.getCapability(Item_Handler) not equal is to what a player expects. If he interacts with the citizens inventory he expects its inventory. You looked at the citizen to interact with his inventory correct?. When you call super you have no idea what inventory gets returned, so the behaviour is undefined. There is no proper reason for calling the super here if we can not supply the inventory of the citizen. So just return false here and leave the super stuff out.